libraries/libstratosphere/source/fssystem/fssystem_local_file_system.cpp

```diff
@@ -0,0 +1,1751 @@
+
+                            /* If we read nothing, we've hit the end of the directory. */
+                            if (nread == 0) {
+                                break;
+                            }
+
+                            /* Determine the number of entries we read. */
+                            auto cur_read_entries = 0;
+                            for (auto pos = 0; pos < nread; pos += ent->d_reclen) {
+                                /* Get the native entry. */
+                                ent = reinterpret_cast<NativeDirectoryEntryType *>(buf + pos);
+
+                                /* If the entry isn't a read target, ignore it. */
+                                if (IsReadTarget(ent)) {
+                                    ++cur_read_entries;
+                                }
+                            }
+
+                            /* If we'll end up reading more than we can fit, allocate a temporary buffer. */
+                            if (read_count + cur_read_entries > max_entries) {
+                                /* Allocate temporary entries. */
+                                m_temp_entries_count = (read_count + cur_read_entries) - max_entries;
+                                m_temp_entries_ofs   = 0;
+
+                                /* TODO: Non-fatal? */
+                                m_temp_entries = fs::impl::MakeUnique<fs::DirectoryEntry[]>(m_temp_entries_count);
+                                AMS_ABORT_UNLESS(m_temp_entries != nullptr);
+                            }
+
+                            /* Iterate received entries. */
+                            for (auto pos = 0; pos < nread; pos += ent->d_reclen) {
+                                /* Get the native entry. */
+                                ent = reinterpret_cast<NativeDirectoryEntryType *>(buf + pos);
+
+                                /* If the entry isn't a read target, ignore it. */
+                                if (!IsReadTarget(ent)) {
+                                    continue;
+                                }
+
+                                /* Decide on the output entry. */
+                                fs::DirectoryEntry *out_entry;
+                                if (read_count < max_entries) {
+                                    out_entry = std::addressof(out_entries[read_count++]);
+                                } else {
+                                    out_entry = std::addressof(m_temp_entries[m_temp_entries_ofs++]);
+                                }
+
+                                /* Setup the output entry. */
+                                {
+                                    std::memset(out_entry->name, 0, sizeof(out_entry->name));
+
+                                    const auto name_len = std::strlen(ent->d_name);
+                                    std::memcpy(out_entry->name, ent->d_name, name_len + 1);
```

Comment:
Uh, bounds check here? Or otherwise leave a comment/static assert here explaining why it's not necessary.

Reply:
I don't think it's necessary because of path max being less than fs::EntryNameLengthMax on both linux and macOS? I can add an error check though, if you think it would be wise.

Reply:
Yeah, that's what I figured, so a nearby `static_assert(PATH_MAX <= fs::EntryNameLengthMax)` would keep anyone like me from asking in the future.